monai/inferers/splitter.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Callable, Iterable, Sequence
+from inspect import _empty, signature
+from typing import Any
+
+import torch
+
+from monai.data.utils import iter_patch_position
+from monai.utils.enums import PytorchPadMode
+from monai.utils.misc import ensure_tuple, ensure_tuple_rep
+from monai.utils.module import look_up_option
+
+__all__ = ["Splitter", "SlidingWindowSplitter"]
+
+
+class Splitter(ABC):
+    """
+    A base class for splitting the inputs into iterable tuple of patches and locations
+    Extend this class to support operations for `PatchInference`, e.g. SlidingPatchSplitter.
+
+    Args:
+        patch_size: the size of patches to be generated.
+        device: the device where the patches are generated.
+    """
+
+    def __init__(self, patch_size: Sequence[int] | int, device: torch.device | str | None = None) -> None:
+        self.patch_size = patch_size
+        self.device = device
+
+    @abstractmethod
+    def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
+        """
+        Split the input image (or batch of images) into patches.
```

Comment:
Could you please help add more description about the expected output of this function?

Thanks.

Reply:
Just did!